src/compiler/checker.ts

```diff
@@ -13512,7 +13512,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
     }
 
     function getConstraintFromIndexedAccess(type: IndexedAccessType) {
-        if (isMappedTypeGenericIndexedAccess(type)) {
+        if (isMappedTypeGenericIndexedAccess(type) || isGenericMappedType(type.objectType)) {

```

Comment:
If the indexed access is allowed - then I think it's always safe to substitute as when the `indexType` is invalid then the error is reported elsewhere anyway.